flink-runtime/src/test/java/org/apache/flink/runtime/leaderelection/DefaultLeaderElectionServiceTest.java

```diff
@@ -1246,6 +1269,73 @@ private void testNonBlockingCall(
+        try (final DefaultLeaderElectionService testInstance =
+                new DefaultLeaderElectionService(
+                        TestingLeaderElectionDriver.newBuilder(
+                                        new AtomicBoolean(false),
+                                        leaderInformationStorage,
+                                        new AtomicBoolean(false))
+                                ::build)) {
+            final String componentId = "test-component";
+            final LeaderElection leaderElection = testInstance.createLeaderElection(componentId);
+
+            // we need the lock to be acquired once for the leadership grant and once for the
+            // revocation
+            final CountDownLatch contenderLockAcquireLatch = new CountDownLatch(2);
+            final OneShotLatch grantReceivedLatch = new OneShotLatch();
+
+            final AtomicBoolean contenderLeadership = new AtomicBoolean(false);
+            final TestingGenericLeaderContender leaderContender =
+                    TestingGenericLeaderContender.newBuilder()
+                            .setPreLockAcquireAction(contenderLockAcquireLatch::countDown)
+                            .setGrantLeadershipConsumer(
+                                    ignoredSessionId -> {
+                                        contenderLeadership.set(true);
+                                        grantReceivedLatch.trigger();
+                                    })
+                            .setRevokeLeadershipRunnable(() -> contenderLeadership.set(false))
+                            .build();
+
+            leaderElection.startLeaderElection(leaderContender);
+
+            final UUID leaderSessionId = UUID.randomUUID();
+            testInstance.onGrantLeadership(leaderSessionId);
+            grantReceivedLatch.await();
+
+            final CompletableFuture<Void> revocationFuture;
+            final CompletableFuture<Void> confirmLeadershipFuture;
+            synchronized (leaderContender.getLock()) {
+                revocationFuture = CompletableFuture.runAsync(testInstance::onRevokeLeadership);
+                contenderLockAcquireLatch.await();
+                confirmLeadershipFuture =
+                        leaderElection.confirmLeadershipAsync(leaderSessionId, "random-address");
+            }
+
+            assertThatFuture(revocationFuture).eventuallySucceeds();
+            assertThatFuture(confirmLeadershipFuture).eventuallySucceeds();
+
+            assertThat(contenderLeadership).isFalse();
+            assertThat(leaderInformationStorage.get().forComponentId(componentId).isPresent())
+                    .as(
+                            "The LeaderInformation is empty because the leadership confirmation succeeded the "
+                                    + "leadership revocation which resulted in no leader information being written out to "
+                                    + "the HA backend.")
+                    .isFalse();
```

Comment:
```suggestion
            assertThat(leaderInformationStorage.get().forComponentId(componentId))
                    .as(
                            "The LeaderInformation is empty because the leadership confirmation succeeded the "
                                    + "leadership revocation which resulted in no leader information being written out to "
                                    + "the HA backend.")
                    .isEmpty();
```
since anyway need to change it, how about this
i guess it should work for both java 8 and 11

Reply:
> since anyway need to change it, how about this

Why do I have to change it once more?

> or isNotPresent()

Good idea. But I'm hesitant to change that to waste another CI run. The currently (admittedly less assertj-like) should work as well

Reply:
>Why do I have to change it once more?

i was referring to the fact that cherry-pick fails ci

>But I'm hesitant to change that to waste another CI run. The currently (admittedly less assertj-like) should work as well

up to you, was just a suggestion

Reply:
👍 I guess, I missed that you commented on it before I pushed a fix.

Sorry for that and thanks for the suggestion, still :-)